src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/TransferOutOfAdaHandler.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event.*;
+
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.DateProvider;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.DetentionStatus;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class TransferOutOfAdaHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final DateProvider dateProvider;
+
+    public TransferOutOfAdaHandler(DateProvider dateProvider) {
+        this.dateProvider = dateProvider;
+    }
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                && callback.getEvent() == TRANSFER_OUT_OF_ADA;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback) {
+
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase =
+                callback
+                        .getCaseDetails()
+                        .getCaseData();
+
+        Optional<YesOrNo> isAcceleratedDetainedAppeal = asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class);
+
+        if (isAcceleratedDetainedAppeal.equals(Optional.of(YesOrNo.YES))) {
```

Comment:
I think at this point we will need to clear existing ADA flags (if they exist on the case) as those will enable events ahead of time compared to the normal appeals flow.

It also might be worth adding in flags into this PR (or 6662) to capture the status of the important events, submit hearing requirements, review hearing requirements & listing which might have happened during the ADA flow before they transferred out.

Reply:
Yes this handler will definitely have more added to it as part of 6662. Elli has summarised behaviour between events and states here: https://tools.hmcts.net/confluence/display/RIA/Overview+tab+progress+bar+after+completion+of+Transfer+out+of+ADA+event

I'm going to put it all under that ticket so that for QA they can do all testing under 1 ticket and any further changes can also be added to that. For these 6586 scenarios this should enough of a skeleton handler

Reply:
Sounds good.